PizzaBuilder/PizzaBuilder/Program.cs

```diff
@@ -0,0 +1,173 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace PizzaBuilder
+{
+    public interface IPizzaBuilder
+    {
+        void Reset();
+        void buildDough();
+        void grabPan();
+        void buildSauce();
+        void buildToppings();
+
+    }
+    class DeepdishPizzaBuilder : IPizzaBuilder
+    {
+        private DeepdishPizza pizza = new DeepdishPizza();
+
+        public DeepdishPizzaBuilder()
+        {
+            this.Reset();
+        }
+        public void Reset()
+        {
+
+        }
+        public void grabPan()
+        {
+            this.pizza.add("deep dish pan");
+        }
+
+        public void buildDough()
+        {
+            this.pizza.add("white flour");
+        }
+
+        public void buildSauce()
+        {
+            this.pizza.add("tomato herb sauce");
+        }
+
+        public void buildToppings()
+        {
+            this.pizza.add("cheese");
+        }
+        public DeepdishPizza getPizza()
+        {
+            DeepdishPizza box = this.pizza;
+            this.Reset();
+            return box;
+        }
+    }
+    public class DeepdishPizza
+    {
+        private List<object> pizzaComponents = new List<object>();
+
+        public void add(string components)
+        {
+            this.pizzaComponents.Add(components);
+        }
+
+        public string makePizza()
+        {
+            string str = "";
+            for (int i = 0; i < this.pizzaComponents.Count; i++)
+            {
+                str += this.pizzaComponents[i] + ",";
+            }
+            return "pizza components:" + str;
+        }
+    }
+    class ThincrustPizzaBuilder : IPizzaBuilder
+    {
+        private ThincrustPizza pizza = new ThincrustPizza();
+
+        public ThincrustPizzaBuilder()
+        {
+            this.Reset();
+        }
+        public void Reset()
```

Comment:
The `reset` method is empty, I think you forgot to implement it..
it should initialize new `ThincrustPizza ` object.